include/alp/structures.hpp

```diff
@@ -621,6 +629,19 @@ namespace alp {
 			using inferred_structures = tuple_cat< std::tuple< Orthogonal >, NonSingular::inferred_structures, OrthogonalColumns::inferred_structures, OrthogonalRows::inferred_structures >::type;
 		};
 
+		template<>
+		struct isInstantiable< Orthogonal, Orthogonal > {
+			template< typename ImfR, typename ImfC >
+			static bool check( const ImfR &imf_r, const ImfC &imf_c ) {
+				// This chek has to further improved.
+				// Orthogonal matrix in the current implemenation
+				// means full-rank square othogonal matrix.
+				// Rectangular matrix orthogonal only by rows (or columns)
```

Comment:
Rectangular matrix cannot possibly be orthogonal along larger dimension, as you cannot have more than n linearly independent vectors of size n. (n+1)^th vector must be linarly dependent on the first n vectors.  This would simply mean that a (m x n) orthogonal matrix is orthogonal along smaller dimension.

Reply:
Correct. What I mean is that in the ontology we have a (currently unused) concept of `OrthogonalRows/Columns` structure which we should use for rectangular cases. Orthogonal in our current spec is specific for square matrices.